pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainSchedulerService.java

```diff
@@ -68,6 +68,7 @@ protected void run()
         @Override
         public void runJob() {
           boolean isFinished = false;
+          boolean returnedErrorBlock = false;
```

Comment:
IMO this file's change can be checked in separately

Reply:
Let's keep them in the same PR since this enforces the behavior that we should call Operator#cancel when there's an error-block returned.

A MailboxReceiveOperator may receive data from multiple mailbox (say 10 of them).

If one of them returns an error-block we bubble up the error-block immediately, so the other mailbox may still be active. So we shouldn't simply close the Operator but rather cancel to signal an early-termination.